### PR TITLE
Feat/sessions

### PR DESCRIPTION
**Pull request by bosi95**
Periodically fetch and update session data from swarm.
Display sessions by recency on homepage.
Filter agenda items by day, stage, like and category.

### CODE REVIEW

File: .env.sample
Comment (ottpeter) [inline .env.sample:1]: is this the gateway feed owner address? there, we need to have overlay address.

File: src/MainRouter.tsx
Comment (ottpeter) [inline src/MainRouter.tsx:1]: how will isBeRunning work together with gateway?

File: src/MainRouter.tsx [hunk 1]
```diff
@@ -12,25 +13,93 @@ import DevconLounge from "./pages/DevconLounge/DevconLounge";
...
+        if (!isBeeRunning) {
+          setBeeRunning(true);
+          console.log("Bee is running");
+        }
+      })
+      .catch(() => {
+        setBeeRunning(false);
+        console.log("Bee stopped running");
+      });
+  }
```
Comment (Kexort) [inline src/MainRouter.tsx:44]: DRY

File: src/MainRouter.tsx [hunk 2]
```diff
@@ -12,25 +13,93 @@ import DevconLounge from "./pages/DevconLounge/DevconLounge";
...
+        const feedRef = feedUpdateRes.reference as string;
+        if (feedRef !== sessionsReference) {
+          setSessionsReference(() => feedRef);
+          console.log("sessions reference updated: ", feedRef);
+        }
+      } catch (e) {
+        console.log("feed download error", e);
+      }
+    }
+  }
```
Comment (Kexort) [inline src/MainRouter.tsx:65]: Consider extracting all swarm related functionality to a separate file

**Review by ottpeter — APPROVED** [review]

File: src/pages/Agenda/Agenda.tsx [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
+import { STAGES_MAP, DAYS_MAP, CATEGORY_FILTERS } from "../../utils/constants";
 
-const Agenda: React.FC = () => {
+interface AgendaProps {
+  sessions: Session[];
+}
+
+const Agenda: React.FC<AgendaProps> = ({ sessions }) => {
+  const [agendaItemsByDay, setAgendaItemsByDay] = useState<JSX.Element[][]>([]);
+  const [activeAgendaItems, setActiveAgendaItems] = useState<JSX.Element[]>([]);
```
Comment (Kexort) [inline src/pages/Agenda/Agenda.tsx:19]: IIRC it is not recommended to use JSXElements in the state

File: src/pages/Agenda/Agenda.tsx [hunk 4]
```diff
@@ -27,86 +36,111 @@ const Agenda: React.FC = () => {
...
+            break;
+          case DAYS_MAP.get("Day 4"):
+            dayIndex = 3;
+            break;
+          default:
+            console.log("unkown day: ", day);
+            break;
+        }
+        const sessionItem = sessions[i];
+        const shortTitle = shortenTitle(sessionItem.title);
```
Comment (Kexort) [inline src/pages/Agenda/Agenda.tsx:69]: Should we do these transformations before we upload the data and keep the frontend light?

File: src/MainRouter.tsx
Comment (bosi95) [reply src/MainRouter.tsx:1]: if the /addresses endpoint can be fetched then there should be no issue

File: src/pages/Agenda/Agenda.tsx (hunk 4 above)
Comment (Kexort) [inline src/pages/Agenda/Agenda.tsx:69]: What is the reason to shorten the title? What is the difference from `text-overflow: ellipsis;`?

File: src/MainRouter.tsx
Comment (Kexort) [reply src/MainRouter.tsx:1]: It cannot be fetched, only the data endpoints are proxied to the node.

File: src/pages/Agenda/Agenda.tsx [hunk 5]
```diff
@@ -27,86 +36,111 @@ const Agenda: React.FC = () => {
...
+      }
+      setActiveAgendaItems(activeItemsArray);
+    }
+  }, [
+    agendaItemsByDay,
+    activeDayTab,
+    activeStageTab,
+    activeAgendaTab,
+    categoryIndex,
+  ]);
```
Comment (Kexort) [inline src/pages/Agenda/Agenda.tsx:124]: You shouldn't build the JSX again when you filter.
The data should be in the state and let React rerender the filtered data.

File: src/utils/constants.ts [hunk 6]
```diff
@@ -1,5 +1,23 @@
 export const BATCH_ID_HEX_LENGTH = 64;
 export const ADDRESS_HEX_LENGTH = 64;
+export const FIVE_MINNUTES = 1000 * 60 * 5;
```
Comment (Kexort) [inline src/utils/constants.ts:3]: typo

File: src/MainRouter.tsx
Comment (bosi95) [reply src/MainRouter.tsx:1]: health check is done via /bytes now

File: src/MainRouter.tsx (hunk 2 above)
Comment (bosi95) [reply src/MainRouter.tsx]: moved to bee.ts

File: src/pages/Agenda/Agenda.tsx (hunk 4 above)
Comment (bosi95) [reply src/pages/Agenda/Agenda.tsx]: uploaded already transformed data

File: src/pages/Agenda/Agenda.tsx (hunk 5 above)
Comment (bosi95) [reply src/pages/Agenda/Agenda.tsx]: fixed

File: src/utils/constants.ts (hunk 6 above)
Comment (bosi95) [reply src/utils/constants.ts]: fixed

File: src/pages/Agenda/Agenda.tsx (hunk 3 above)
Comment (bosi95) [reply src/pages/Agenda/Agenda.tsx]: fixed

**Review by Kexort — APPROVED** [review]

**Pull request merged**